Main.js

```diff
@@ -223,28 +223,30 @@ requirejs([
 				$('#accordion').accordion();
 			}
 		});
-		//$("#accordion").accordion();
-		$('#enterPassword').dialog({
-			modal: true,
-			buttons: {
-				Ok: function() {
-					currentServer.user.setPasswd($('#password').prop('value')) ;
-					Interface.login();
-					$('#enterPassword').dialog('close');
-				},
-				Cancel: function(){
-					$.cookie('userId', undefined);
-					$.cookie('passwd', undefined);
-					$('#enterPassword').dialog('close');
-				}
-			},
-			autoOpen: false,
-			close: function(){this.title = 'Введите пароль';}
-		});
-		$('#enterPassword').on('keyup', function(e){
-		  if (e.keyCode == 13) {
-			$(this).dialog( "option", "buttons" )['Ok']();
-		  }});
+
+        $('#login-dialog').dialog({
+          autoOpen: false,
+          height: 300,
+          width: 350,
+          modal: true,
+          buttons: [{
+            text: 'Войти',
+            click: function () {
+              var $self = $(this);
+              var login = $('#login', this).val();
+              var passwd = $('#password', this).val();
+              Interface.login(login, passwd, function () {
```

Comment:
In case of failure, `login` calls `showLoginDialog` without touching `callback`.
This probably works in practice, but still I'm uncomfortable about such code.